addons/website_sale/static/src/js/website_sale.editor.js

```diff
@@ -236,29 +356,48 @@ options.registry.WebsiteSaleProductsItem = options.Class.extend({
     /**
      * @override
      */
-    _renderCustomWidgets: async function (uiFragment) {
-        const checkboxes = [];
+    async _renderCustomXML(uiFragment) {
+        const $select = $(uiFragment.querySelector('we-select'));
         return this._rpc({
-            model: 'product.style',
+            model: 'product.ribbon',
             method: 'search_read',
-        }).then(async data => {
-            for (var k in data) {
-                const checkboxWidget = this._registerUserValueWidget('we-checkbox', this, data[k]['name'], {
-                    dataAttributes: {
-                        'style': data[k]['id'],
-                        'selectClass': data[k]['html_class'],
-                    },
-                });
-                checkboxes.push(checkboxWidget);
-                await checkboxWidget.appendTo(document.createDocumentFragment());
-            }
-        }).then(() => {
-            const menuEl = uiFragment.querySelector('[name="style"]');
-            for (const checkbox of checkboxes) {
-                menuEl.appendChild(checkbox.el);
+            fields: ['id', 'name', 'color', 'text_color'],
+        }).then(async ribbons => {
+            this.ribbons = Object.fromEntries(ribbons.map(r => [r.id, r]));
+            for (const ribbon of ribbons) {
+                $select.append($(`<we-button data-set-ribbon="${ribbon.id}">${ribbon.name}</we-button>`));
```

Comment:
If you are using jQuery then `append(string)` will be enough, instead of `append($(string))`